### PR TITLE
Set default axes for N-dimensional fft APIs

### PR DESCRIPTION
**Pull request by yasahi-hpc**
In this PR, we set default axes for ND fft APIs. 
For example, for 2D Views, default `axes = {-2, -1}`.

```C++
using axes_type = KokkosFFT::axis_type<2>;
axes_type axes  = {-2, -1};
KokkosFFT::fftn(execution_space(), x, out);
KokkosFFT::fftn(execution_space(), x, out, axes);
```

### CODE REVIEW

**Review by cedricchevalier19 — DISMISSED** [review]
Can you be more explicit in the PR title? `ND` is not very clear. "N-D" or "N-dimensional" is easier to understand.

File: fft/unit_test/Test_Transform.cpp
Comment (yasahi-hpc) [reply fft/unit_test/Test_Transform.cpp:2326]: Sure. I will add them

File: common/unit_test/Test_Utils.cpp
Comment (yasahi-hpc) [reply common/unit_test/Test_Utils.cpp:493]: Maybe better to test a real use case: make an index sequence from Views. I will do that.

File: common/src/KokkosFFT_utils.hpp
Comment (tpadioleau) [inline common/src/KokkosFFT_utils.hpp:138]: I think `make_sequence_array` can now be removed

File: common/src/KokkosFFT_utils.hpp
Comment (tpadioleau) [inline common/src/KokkosFFT_utils.hpp:152]:
```suggestion
template <typename IntType, std::size_t N, IntType start>
constexpr std::array<IntType, N> index_sequence() {
  static_assert(std::is_integral_v<IntType> && std::is_signed_v<IntType>,
                "index_sequence: IntType must be a signed integer type.");
  std::array<IntType, N> sequence{};
  for (std::size_t i = 0; i < N; ++i) {
    sequence[i] = start + static_cast<IntType>(i);
  }
  return sequence;
}
```

I don't think we need to make a distinction between the two types ?

**Review by tpadioleau — CHANGES_REQUESTED** [review]

File: common/src/KokkosFFT_utils.hpp
Comment (yasahi-hpc) [reply common/src/KokkosFFT_utils.hpp]: True. Fixed accordingly

File: common/src/KokkosFFT_utils.hpp
Comment (yasahi-hpc) [reply common/src/KokkosFFT_utils.hpp]: I have removed this

File: fft/unit_test/Test_Transform.cpp
Comment (yasahi-hpc) [reply fft/unit_test/Test_Transform.cpp]: I added new fully defaulted cases.

**Comment by yasahi-hpc** [comment]
> Can you be more explicit in the PR title? `ND` is not very clear. "N-D" or "N-dimensional" is easier to understand.

Thank you for your review. I have fixed accordingly. May I have your second review?

**Review by tpadioleau — APPROVED** [review]
Seems good to me, thanks for this feature

**Comment by yasahi-hpc** [comment]
Sure, I will merge this. Thank you for your help

**Pull request merged**
